unit-tests/src/test/scala/scala/scalanative/native/SizeofSuite.scala

```diff
@@ -75,9 +75,9 @@ object SizeofSuite extends tests.Suite {
                  CStruct3[Int, Int, CStruct4[Byte, Int, Short, Byte]]]] == 40)
   }
 
-  type _32   = Nat.Digit[Nat._3, Nat._2]
-  type _128  = Nat.Digit[Nat._1, Nat.Digit[Nat._2, Nat._8]]
-  type _1024 = Nat.Digit[Nat._1, Nat.Digit[Nat._0, Nat.Digit[Nat._2, Nat._4]]]
+  type _32   = Nat.Digit2[Nat._3, Nat._2]
+  type _128  = Nat.Digit3[Nat._1, Nat._2, Nat._8]
+  type _1024 = Nat.Digit4[Nat._1, Nat._0, Nat._2, Nat._4]
```

Comment:
/cc @rwhaling This is the new Nat encoding that I mentioned at some point. IMO it looks nicer actually. 😄

Reply:
With this change, sizeof is always constant folded away and is zero cost despite being completely implemented in the library code.